### ISSUE TITLE
CodeQL should support SARIF message string lookup in uploaded logs

### ISSUE THREAD

**Issue by EasyRhinoMSFT** [github/codeql#11512]
The SARIF format supports the notion of indirect string lookups in its `message` object. Rather than directly providing a string in the `message.text` property, the indirect approach allows a specific set of other objects within the SARIF log to refer to the `message` object using its `id` property.

Currently, the CodeQL web service returns an error response when a log is uploaded which contains any `message` objects that don't provide a value in the `text` property:
`Error: Code Scanning could not process the submitted SARIF file: expected a result message`

The formal description of this feature and the decision tree for resolving the message string can be found here: [https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317468](https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317468)

**Comment by cannist** [comment]
That is correct. At the moment the code scanning service only supports result messages in the `message.text` property. We have documented that [here](https://docs.github.com/en/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#result-object).

May I ask which tool is producing the sarif file that is getting rejected?

**Comment by EasyRhinoMSFT** [comment]
I should be more clear: this is a change request, not a bug report.

The producer in this case is BinSkim, but there are others that also use this approach.

**Comment by cannist** [comment]
Ah, ok. I am sorry that I misunderstood you. I have passed this on to our Product Manager.

**Comment by dorothymitchell** [comment]
Thanks for the feedback @EasyRhinoMSFT. We've made an issue to track this and record if any other users request this feature. I'll let you know if we prioritize working on this improvement! Thanks!

**Comment by dronkoff** [comment]
Hit the same issue with https://github.com/Azure/template-analyzer results.
Any progress?

**Comment by AlonaHlobina** [comment]
Hi @dronkoff,  
Unfortunately, due to our current priorities and limited team capacity, we haven’t been able to prioritize this just yet. I appreciate your understanding and patience.

**Comment by dronkoff** [comment]
Hi @AlonaHlobina , thanks for quick response. I reformatted the messages with a simple script: https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/blob/main/tools/Reformat-Sarif.ps1 and it works pretty well https://github.com/ssc-spc-ccoe-cei/azure-guardrails-solution-accelerator/blob/main/.github/workflows/Bicep_code_scan.yml . 
However, it turns out that there is also an issue with relative file location (location.uriBaseId) support as well https://docs.oasis-open.org/sarif/sarif/v2.1.0/os/sarif-v2.1.0-os.html#_Toc34317431 .
